components/base/sensorcontrolled/velocities.go

```diff
@@ -84,6 +95,18 @@ func (sb *sensorBase) setupControlLoop(linear, angular control.PIDConfig) error
 	return nil
 }
 
+func (sb *sensorBase) waitForTuning() {
+	for {
+		if !control.TunedPIDVals[0].NeedsAutoTuning() && !control.TunedPIDVals[1].NeedsAutoTuning() {
```

Comment:
gutils.SelectContextOrWait(ctx, X*time.Second) so we don't get stuck in the loop